scripts/usb_reset.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python
+#
+# Copyright (C) Citrix Systems Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published
+# by the Free Software Foundation; version 2.1 only. #
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# ./usb_reset.py 1-2
+# reset USB device 1-2
+
+import fcntl
+import xcp.logger as log
+import logging
+import re
+import sys
+import traceback
+
+#USBDEVFS_RESET _IO('U', 20)
+USBDEVFS_RESET = (ord('U') << 8) + 20
+
+
+def read_int(path):
+    with open(path) as f:
+        return int(f.readline())
+
+
+def reset_device(bus, dev):
+    with open("/dev/bus/usb/{0:03d}/{1:03d}".format(bus, dev), 'w') as f:
+        fcntl.ioctl(f, USBDEVFS_RESET, 0)
+
+
+if __name__ == "__main__":
+    log.logToSyslog(level=logging.DEBUG)
+
+    if len(sys.argv) != 2:
+        log.debug("usage: {} device_node".format(sys.argv[0]))
```

Comment:
Usage errors should be logged to stderr using

    sys.exit("usage: {} device_node".format(sys.argv[0]))

Reply:
The usage could also include the expected syntax of the argument. `usage: {} d-d[.d]`. I understand that this is not the precise syntax but without any hint a user is even more on their own. Or maybe give a hint that this is looking into `/sys/bus/usb/devices/` where these are found? I think there could be even an argument to be made to accept the full path to this device.

Reply:
@lindig Actually, I am coding another script(usb_scanner.py) to get all the proper USB info. So we can keep it simple here.